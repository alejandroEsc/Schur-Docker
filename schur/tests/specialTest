#!/bin/bash

# params = SCHUROOT(=SCHURLIB) builddir 

SCHURLIB=$1
export SCHURLIB
SCHUR=$2
COLUMNS=67
export COLUMNS
SUPPR=0

echo "specialTests... may be long..."
if [ ! -e testrule.schur ]; then
	cp $1/tests/test*.schur .
	SUPPR=1
fi
${SCHUR} < $1/tests/specialTest.cmds | ./trim > specialTest.res
if diff -w specialTest.res $1/tests/specialTest.expected > specialTest.diff ; then
	rm specialTest.{res,diff}
	if [ A${SUPPR} == A1 ]; then
	       	rm -f test*.schur
	fi
else
	echo "There is a problem in specialTests"
	cat specialTest.diff
	echo ": "`grep '<' specialtest.diff | wc -l`' differences (?)'
	exit 1
fi

