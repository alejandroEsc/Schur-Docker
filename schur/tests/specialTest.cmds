sfn
o 0,0
o 0,321
o 1,321
o321,0
o321,1
o21,-2
o-21,-21
o !100 , !100
o !199 , 1
conj last
conj 1^!200
conj last
mult_p 20 !10
setsv1 o21,-21
sv1
o sv1,21
add o21,-21,o21,21
sk sv1,21
sk sv1,-21
sk last,21
sk 321,a
sk 321,b
sk 321,c 
sk 321,d 
sk 321,e
sk 321,f
sk 321,g
sk 321,h 
sk 321,l
sk 321,m
1111+&
2222+3333+&
4444+5555+6666+7777+8888+9999&
4444+5555+6666+7777+8888+9999+1+2+3+4+5+6+7+8+9+11+22+33+44+55+66+77+88+99+111+222+333+444
setsv1 &
1111+2222+3333+4444+5555+6666+7777+8888+9999+111+222+333+444+555+666+777+888+999+11111+22222+33333+44444+55555+66666+77777+88888+99999+111111+222222+333333+444444+555555+666666+777777+888888+999999+1
sv1
REM return to DPrepMode, to try examples from the manual
exit
gr u20
dim[1^!14 ]
br5 2 10 gr1[1^!14 ]
gr2 su2 su10
std last
au gr1so3,last
setv1,last
v1
gr su10
br8 2 8 gr1[2^7]
uonediv10gr1uoneadd28gr1last
gr su8
br8 4 4 gr1[2^5]
uonediv8gr1uoneadd40gr1last
rem ../tests needed for make distcheck to work
readfn 1 'testrule.schur'
fn1
readfn 2 'testrule2.schur'
fn2
readfn 3 'testrule3.schur'
fn3
end
