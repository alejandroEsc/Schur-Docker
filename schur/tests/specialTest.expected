DPrep Mode (with function)
Schur Function Mode
{0}
{321}
{421} + {3^2 1} + {32^2} + {321^2}
{321}
{421} + {3^2 1} + {32^2} + {321^2}
- {41} - {32} - {31^2} - {2^2 1}
{42} + {41^2} + {3^2} + 2{321} + {31^3} + {2^3} + {2^2 1^2}
{200 } + {199 1} + {198 2} + {197 3} + {196 4} + {195 5} + {194 6}
+ {193 7} + {192 8} + {191 9} + {190 10 } + {189 11 } + {188 12 }
+ {187 13 } + {186 14 } + {185 15 } + {184 16 } + {183 17 }
+ {182 18 } + {181 19 } + {180 20 } + {179 21 } + {178 22 }
+ {177 23 } + {176 24 } + {175 25 } + {174 26 } + {173 27 }
+ {172 28 } + {171 29 } + {170 30 } + {169 31 } + {168 32 }
+ {167 33 } + {166 34 } + {165 35 } + {164 36 } + {163 37 }
+ {162 38 } + {161 39 } + {160 40 } + {159 41 } + {158 42 }
+ {157 43 } + {156 44 } + {155 45 } + {154 46 } + {153 47 }
+ {152 48 } + {151 49 } + {150 50 } + {149 51 } + {148 52 }
+ {147 53 } + {146 54 } + {145 55 } + {144 56 } + {143 57 }
+ {142 58 } + {141 59 } + {140 60 } + {139 61 } + {138 62 }
+ {137 63 } + {136 64 } + {135 65 } + {134 66 } + {133 67 }
+ {132 68 } + {131 69 } + {130 70 } + {129 71 } + {128 72 }
+ {127 73 } + {126 74 } + {125 75 } + {124 76 } + {123 77 }
+ {122 78 } + {121 79 } + {120 80 } + {119 81 } + {118 82 }
+ {117 83 } + {116 84 } + {115 85 } + {114 86 } + {113 87 }
+ {112 88 } + {111 89 } + {110 90 } + {109 91 } + {108 92 }
+ {107 93 } + {106 94 } + {105 95 } + {104 96 } + {103 97 }
+ {102 98 } + {101 99 } + {100^2}
{200 } + {199 1}
{21^198} + {1^200 }
{200 }
{1^200 }
{200 }
- {42} - {41^2} - {3^2} - 2{321} - {31^3} - {2^3} - {2^2 1^2}
- {63} - 2{621} - {61^3} - 2{54} - 6{531} - 4{52^2} - 6{521^2}
- 2{51^4} - 4{4^2 1} - 8{432} - 9{431^2} - 9{42^2 1} - 6{421^3}
- {41^5} - 2{3^3} - 8{3^2 21} - 4{3^2 1^3} - 4{32^3} - 6{32^2 1^2}
- 2{321^4} - 2{2^4 1} - {2^3 1^3}
zero
- 4{3} - 10{21} - 4{1^3}
4{3} + 10{21} + 4{1^3}
10{0}
{321} - {31} - {2^2} - {21^2} + {2} + {1^2}
{321} + {31} + {2^2} + {21^2} + {2} + {1^2}
{321} - {31} - {2^2} - {21^2} + {2} + {1^2}
{321} + {31} + {2^2} + {21^2} + {2} + {1^2}
{321} - {32} - {31^2} + {3} - {2^2 1} + 2{21} - {2} + {1^3}
- {1^2} - {1} + {0}
{321} + {32} + {31^2} + 2{31} + {3} + {2^2 1} + 2{2^2} + 2{21^2}
+ 4{21} + 3{2} + {1^3} + 3{1^2} + 3{1} + {0}
{321} + {32} + {31^2} - {3} + {2^2 1} - 2{21} - {2} - {1^3}
- {1^2} + {1} + {0}
{321} - {32} - {31^2} + 2{31} - {3} - {2^2 1} + 2{2^2} + 2{21^2}
- 4{21} + 3{2} - {1^3} + 3{1^2} - 3{1} + {0}
{321} - {32} - {31^2} + {31} - {2^2 1} + {2^2} + {21^2} - {21}
{321} + {32} + {31^2} + {31} + {2^2 1} + {2^2} + {21^2} + {21}
{1^4} + {2^4} + {3^4} + {4^4} + {5^4} + {6^4} + {7^4} + {8^4}
+ {9^4}
{4^4} + {5^4} + {6^4} + {7^4} + {8^4} + {9^4} + {1} + {2} + {3}
+ {4} + {5} + {6} + {7} + {8} + {9} + {1^2} + {2^2} + {3^2}
+ {4^2} + {5^2} + {6^2} + {7^2} + {8^2} + {9^2} + {1^3} + {2^3}
+ {3^3} + {4^3}
{9^6} + {9^5} + {9^4} + {9^3} + {8^6} + {8^5} + {8^4} + {8^3}
+ {7^6} + {7^5} + {7^4} + {7^3} + {6^6} + {6^5} + {6^4} + {6^3}
+ {5^6} + {5^5} + {5^4} + {5^3} + {4^6} + {4^5} + {4^4} + {4^3}
+ {3^6} + {3^5} + {3^4} + {3^3} + {2^6} + {2^5} + {2^4} + {2^3}
+ {1^6} + {1^5} + {1^4} + {1^3} + {1}
DPrep Mode (with function)
Group is U(20)
Dimension = 38760
Groups are U(2) * U(10)
{10 4}{2^4 1^6} + {95}{2^5 1^4} + {86}{2^6 1^2} + {7^2}{2^7}
Groups are SU(2) * SU(10)
{6}{1^4} + {4}{2^5 1^4} + {2}{2^6 1^2} + {0}{2^7}
Groups are SO(3) * SU(10)
[3]{1^4} + [2]{2^5 1^4} + [1]{2^6 1^2} + [0]{2^7}
[3]{1^4} + [2]{2^5 1^4} + [1]{2^6 1^2} + [0]{2^7}
Group is SU(10)
Groups are U(1) * SU(2) * SU(8)
{12 }{0}{2^5} + {2}{1}{2^5 1} + {-8}{2}{2^6} + {-8}{0}{2^5 1^2}
+ {-18  }{1}{2^6 1} + {-28  }{0}{2^7}
{4}{0}{2^5} + {3}{1}{2^5 1} + {2}{2}{2^6} + {2}{0}{2^5 1^2}
+ {1}{1}{2^6 1} + {0}{0}{2^7}
Group is SU(8)
Groups are U(1) * SU(4) * SU(4)
{24 }{0}{2} + {16 }{1^3}{21} + {8}{2^3}{2^2} + {8}{1^2}{21^2}
+ {0}{2^2 1}{2^2 1} + {0}{1}{1} + {-8}{2^2}{2^3} + {-8}{21^2}{1^2}
+ {-16  }{21}{1^3} + {-24  }{2}{0}
{8}{0}{2} + {7}{1^3}{21} + {6}{2^3}{2^2} + {6}{1^2}{21^2}
+ {5}{2^2 1}{2^2 1} + {5}{1}{1} + {4}{2^2}{2^3} + {4}{21^2}{1^2}
+ {3}{21}{1^3} + {2}{2}{0}
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
Groups are Sp(6) * Sp(6)
Dimension = 12096
Group is Sp(6)
<52> + <51^2> + <5> + <43> + 2<421> + 3<41> + <3^2 1> + <32^2>
+ 3<32> + 3<31^2> + 2<3> + 2<2^2 1> + 3<21> + <1^3> + <1>
Dimension = 12096
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
Group is O(32)
Dimension = 10880
Groups are Sp(8) * Sp(4)
<3><21> + <3><1> + <21><3> + <21><21> + 2<21><1> + <1^3><21>
+ <1^3><1> + <1><3> + 2<1><21> + 2<1><1>
Dimension = 10880
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
=-
Group is U(64)
Dimension = 87360
Groups are U(8) * U(8) * U(8) * U(8) * U(8) * U(8) * U(8)
Groups are U(8) * U(8) * U(8) * U(8) * U(8) * U(8)
Groups are U(8) * U(8) * U(8) * U(8) * U(8)
Groups are U(8) * U(8) * U(8) * U(8)
Groups are U(8) * U(8) * U(8)
Groups are U(8) * U(8)
Group is U(8)
Group is S(8)
2{8} + 19{71} + 53{62} + 54{61^2} + 69{53} + 156{521} + 79{51^3}
+ 33{4^2} + 160{431} + 126{42^2} + 195{421^2} + 70{41^4}
+ 91{3^2 2} + 118{3^2 1^2} + 142{32^2 1} + 124{321^3} + 37{31^5}
+ 25{2^4} + 51{2^3 1^2} + 35{2^2 1^4} + 10{21^6} + {1^8}
Dimension = 87360
